contracts/NFT.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.3;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+contract NFT is ERC721Enumerable, Ownable {
+    using Counters for Counters.Counter;
+    address private immutable _self;
+    string private baseURI;
+
+    // This address is used for if current owner want to renounceOwnership, it will always be the same address
+    address private constant fixedOwnerAddress =
+        0x1156B992b1117a1824272e31797A2b88f8a7c729;
```

Comment:
Please make this a constructor argument instead of a constant. It should be `immutable` then. It makes more sense that way for templates.